contracts/FastBreak.cdc

```diff
@@ -0,0 +1,682 @@
+        }
+
+        /// Create a game of Fast Break
+        ///
+        pub fun createFastBreakGame(
+            id: String,
+            name: String,
+            fastBreakRunID: String,
+            isPublic: Bool,
+            submissionDeadline: UInt64,
+            numPlayers: UInt64
+        ) {
+            let fastBreakGame: FastBreak.FastBreakGame = FastBreak.FastBreakGame(
+                id: id,
+                name: name,
+                fastBreakRunID: fastBreakRunID,
+                isPublic: isPublic,
+                submissionDeadline: submissionDeadline,
+                numPlayers: numPlayers
+            )
+            FastBreak.fastBreakGameByID[fastBreakGame.id] = fastBreakGame
+            emit FastBreakGameCreated(
+                id: fastBreakGame.id,
+                name: fastBreakGame.name,
+                fastBreakRunID: fastBreakGame.fastBreakRunID,
+                isPublic: fastBreakGame.isPublic,
+                submissionDeadline: fastBreakGame.submissionDeadline,
+                numPlayers: fastBreakGame.numPlayers
+            )
+        }
+
+        /// Add a Fast Break Statistic to a game of Fast Break during game creation
+        ///
+        pub fun addStatToFastBreakGame(fastBreakGameID: String, name: String, type: String, valueNeeded: UInt64) {
+            let fastBreakGame: &FastBreak.FastBreakGame = (&FastBreak.fastBreakGameByID[fastBreakGameID] as &FastBreak.FastBreakGame?)!
+            let fastBreakStat : FastBreak.FastBreakStat = FastBreak.FastBreakStat(
+                name: name,
+                type: type,
+                valueNeeded: valueNeeded
+            )
+
+            fastBreakGame.addStat(stat: fastBreakStat)
+            emit FastBreakGameStatAdded(
+                fastBreakGameID: fastBreakGame.id,
+                name: fastBreakStat.name,
+                type: fastBreakStat.type,
+                valueNeeded: fastBreakStat.valueNeeded
+            )
+        }
+
+        /// Update the status of a Fast Break
+        ///
+        pub fun updateFastBreakGame(id: String, status: String, winner: Address?) {
+            let fastBreakGame: &FastBreak.FastBreakGame = (&FastBreak.fastBreakGameByID[id] as &FastBreak.FastBreakGame?)!
+
+            fastBreakGame.update(status: status, winner: winner)
+            emit FastBreakGameStatusChange(id: fastBreakGame.id, newStatus: fastBreakGame.status)
+        }
+
+        /// Submit a Fast Break on behalf of a wallet
+        ///
+        pub fun submitFastBreak(wallet: Address, submission: FastBreak.FastBreakSubmission) {
+            let fastBreakGame: &FastBreak.FastBreakGame = (&FastBreak.fastBreakGameByID[submission.fastBreakGameID] as &FastBreak.FastBreakGame?)!
+            fastBreakGame.submitFastBreak(submission: submission)
+        }
+
+        /// Updates the submission scores of a Fast Break
+        ///
+        pub fun updateFastBreakScore(fastBreakGameID: String, wallet: Address, points: UInt64, win: Bool) {
```

Comment:
Any considering needed for the new tie-break rule of lowest serial numbers? Or is that something that will be reconciled off-chain?

Reply:
Good call. I think that's a good future function because right now, the off-chain version does not allow this because if anyone submitted to the former stat, their record would get stranded.

To unblock that, I need to add a new state to the game like "DRAFT" and allow changes during this period while also restricting user submission when the Fast Break is in this state.

At this point, we don't support this off-chain so need to have the on-chain stats be immutable as well.

Perhaps a function to remove a Fast Break entirely on-mistake would be nice to have.